src/ring_theory/witt_vector/compare.lean

```diff
@@ -0,0 +1,205 @@
+    (eq_of_le_of_cast_pow_eq_zero p n (zmod p))
+
+local attribute [instance] char_p_zmod
+variable (n)
+
+/--
+Since `truncated_witt_vector p n (zmod p)` is a finite ring
+with characteristic and cardinality `p^n`,
+it is isomorphic to `zmod (p^n)`.
+-/
+def zmod_equiv_trunc : zmod (p^n) ≃+* truncated_witt_vector p n (zmod p) :=
+zmod.ring_equiv (truncated_witt_vector p n (zmod p)) (card_zmod _ _)
+
+lemma zmod_equiv_trunc_apply {x : zmod (p^n)} :
+  zmod_equiv_trunc p n x =
+  zmod.cast_hom (by refl) (truncated_witt_vector p n (zmod p)) x :=
+rfl
+
+lemma commutes {m : ℕ} (hm : n ≤ m) :
+  (truncate hm).comp (zmod_equiv_trunc p m).to_ring_hom =
+    (zmod_equiv_trunc p n).to_ring_hom.comp (zmod.cast_hom (pow_dvd_pow p hm) _) :=
+ring_hom.ext_zmod _ _
+
+lemma commutes' {m : ℕ} (hm : n ≤ m) (x : zmod (p^m)) :
+  truncate hm (zmod_equiv_trunc p m x) =
+    zmod_equiv_trunc p n (zmod.cast_hom (pow_dvd_pow p hm) _ x) :=
+show (truncate hm).comp (zmod_equiv_trunc p m).to_ring_hom x = _,
+by rw commutes _ _ hm; refl
+
+lemma commutes_symm' {m : ℕ} (hm : n ≤ m) (x : truncated_witt_vector p m (zmod p)) :
+  (zmod_equiv_trunc p n).symm (truncate hm x) =
+    zmod.cast_hom (pow_dvd_pow p hm) _ ((zmod_equiv_trunc p m).symm x) :=
+begin
+  apply (zmod_equiv_trunc p n).injective,
+  rw ← commutes',
+  simp
+end
+
+lemma commutes_symm {m : ℕ} (hm : n ≤ m)  :
+  (zmod_equiv_trunc p n).symm.to_ring_hom.comp (truncate hm) =
+    (zmod.cast_hom (pow_dvd_pow p hm) _).comp (zmod_equiv_trunc p m).symm.to_ring_hom :=
+by ext; apply commutes_symm'
+
+end iso
+
+end truncated_witt_vector
+
+namespace witt_vector
+open truncated_witt_vector
+
+variables {p : ℕ} [hp : fact p.prime]
+include hp
+
+local notation `𝕎` := witt_vector p -- type as `\bbW`
+
+variables (p)
+
+/--
+`to_zmod_pow` is a family of compatible ring homs. We get this family by composing
+`truncated_witt_vector.zmod_equiv_trunc` (in right-to-left direction)
+with `witt_vector.truncate`.
+-/
+def to_zmod_pow (k : ℕ) : 𝕎 (zmod p) →+* zmod (p ^ k) :=
+(zmod_equiv_trunc p k).symm.to_ring_hom.comp (truncate k)
+
+lemma to_zmod_pow_compat (m n : ℕ) (h : m ≤ n) :
+  (zmod.cast_hom (pow_dvd_pow p h) (zmod (p ^ m))).comp ((λ (k : ℕ), to_zmod_pow p k) n) =
+    (λ (k : ℕ), to_zmod_pow p k) m :=
+calc (zmod.cast_hom _ (zmod (p ^ m))).comp
+      ((zmod_equiv_trunc p n).symm.to_ring_hom.comp (truncate n)) =
+  ((zmod_equiv_trunc p m).symm.to_ring_hom.comp
+    (truncated_witt_vector.truncate h)).comp (truncate n) :
+  by rw [commutes_symm, ring_hom.comp_assoc]
+... = (zmod_equiv_trunc p m).symm.to_ring_hom.comp (truncate m) :
+  by rw [ring_hom.comp_assoc, truncate_comp_witt_vector_truncate]
+
+/--
+`to_padic_int` lifts `to_zmod_pow` to a ring hom to `ℤ_[p]` using `padic_int.lift`, the universal
+property of `ℤ_[p]`.
+-/
+def to_padic_int : 𝕎 (zmod p) →+* ℤ_[p] :=
+-- I think the family should be an explicit argument of `lift`,
+-- for increased readability.
+padic_int.lift (λ m n h, to_zmod_pow_compat p m n h)
```

Comment:
I don't really know what the comment means. Do you mean you don't want to write
```
padic_int.lift $ to_zmod_pow_compat p
```
?

Reply:
Thanks for the suggestion. I've fixed the code.